play-json/shared/src/main/scala/Json.scala

```diff
@@ -393,15 +398,15 @@ object Json extends JsonFacade {
    * val r: Reads[Foo] = Json.configured.reads[Foo]
    * }}}
    */
-  def configured[C <: JsonConfiguration.Aux[_ <: MacroOptions]](implicit config: C) = new WithOptions[config.Opts]()
+  def configured[Opts <: MacroOptions](implicit config: JsonConfiguration.Aux[Opts]) = new WithOptions[Opts](config)
```

Comment:
Not sure what the go with the old type was, but what we really needed was to return a `WithOptions` that has the same `MacroOptions` type as the passed in configuration, this does that.